Userland/js.cpp

```diff
@@ -159,19 +159,96 @@ static void print(JS::Value value)
     putchar('\n');
 }
 
+String sanitize_repl_inputs(String input)
+{
+    StringBuilder result;
+    for (char c : input) {
+        if (c != ' ' && c != '\n') {
+            result.append(c);
+        }
+    }
+    return result.to_string();
+}
+
+void handle_repl_commands(String part, JS::Interpreter& interpreter)
+{
+    enum ReplCommands {
+        Exit,
+        Load,
+        Help,
+        Unhandled
+    };
+
+    ReplCommands command = ReplCommands::Unhandled;
+
+    auto typed_command = sanitize_repl_inputs(part.split(' ').first());
+    if (typed_command == ".exit")
+        command = ReplCommands::Exit;
+    if (typed_command == ".load")
+        command = ReplCommands::Load;
+    if (typed_command == ".help")
+        command = ReplCommands::Help;
+
+    switch (command) {
+    case ReplCommands::Exit:
+        exit(0);
+        break;
+    case ReplCommands::Unhandled:
+        printf("Unhandled repl command. Type .help to see accepted commands\n");
+        break;
+    case ReplCommands::Help:
+        printf("REPL commands:\n");
+        printf("    .help: display this menu\n");
+        printf("    .exit: exit the REPL\n");
+        printf("    .load: load a javascript file(s) into the repl. Accepts a space seperated list. For example: .load js/one.js js/two.js\n");
+        break;
+    case ReplCommands::Load:
+        auto split = part.split(' ');
+        for (int i = 1; i < static_cast<int>(split.size()); i++) {
+            String file_name = sanitize_repl_inputs(split[i]);
+            auto file = Core::File::construct(file_name);
+            if (!file->open(Core::IODevice::ReadOnly)) {
+                printf("Failed to open %s: %s\n", file_name.characters(), file->error_string());
+            }
+            auto file_contents = file->read_all();
+
+            StringView source;
+            if (file_contents.size() >= 2 && file_contents[0] == '#' && file_contents[1] == '!') {
+                size_t i = 0;
+                for (i = 2; i < file_contents.size(); ++i) {
+                    if (file_contents[i] == '\n')
+                        break;
+                }
+                source = StringView((const char*)file_contents.data() + i, file_contents.size() - i);
```

Comment:
Maybe this part should actually be its own function since it's just the exact same thing as what runs to strip the shebang when the js program is passed a file? What do you think @awesomekling

Reply:
Yes, that's a great idea :)